### PR TITLE
Show files of supplementary variables explicitly in log

### LINKED ISSUE #2302: No explicit logging of supplementary variable file paths
**Is your feature request related to a problem? Please describe.**

Currently, file paths of supplementary variables are logged together with their parent variable:

https://github.com/ESMValGroup/ESMValCore/blob/79455c1dfb4c29e801472cadccd723027997cc88/esmvalcore/_recipe/recipe.py#L252-L269

While this is usually not a problem for CMIP datasets where the variable name appears in the path, this might be problematic for native model output where the variable name is not in the path, which might lead to confusing output.

It would be good if the log specifically mentioned which file belongs to which variable.

**Would you be able to help out?**
Yes!

### PR DESCRIPTION
**Pull request by schlunma**
## Description

This PR makes sure that supplementary variables are shown explicitly in the debug log.

Before:

```bash
2024-01-17 10:30:37,879 UTC [504383] DEBUG   esmvalcore._recipe.recipe:263 Using input files for variable tas of dataset CMIP6 BCC-ESM1:
/work/bd0854/DATA/ESMValTool2/download/CMIP6/CMIP/BCC/BCC-ESM1/historical/r1i1p1f1/Amon/tas/gn/v20181214/tas_Amon_BCC-ESM1_historical_r1i1p1f1_gn_185001-201412.nc
/work/bd0854/DATA/ESMValTool2/CMIP6_DKRZ/CMIP/BCC/BCC-ESM1/1pctCO2/r1i1p1f1/fx/areacella/gn/v20190613/areacella_fx_BCC-ESM1_1pctCO2_r1i1p1f1_gn.nc
/work/bd0854/DATA/ESMValTool2/CMIP6_DKRZ/CMIP/BCC/BCC-ESM1/1pctCO2/r1i1p1f1/fx/sftlf/gn/v20190613/sftlf_fx_BCC-ESM1_1pctCO2_r1i1p1f1_gn.nc
```

After:

```bash
2024-01-17 10:19:03,774 UTC [447883] DEBUG   esmvalcore._recipe.recipe:272 Using input files for variable tas of dataset CMIP6 BCC-ESM1:
  /work/bd0854/DATA/ESMValTool2/download/CMIP6/CMIP/BCC/BCC-ESM1/historical/r1i1p1f1/Amon/tas/gn/v20181214/tas_Amon_BCC-ESM1_historical_r1i1p1f1_gn_185001-201412.nc
with files for supplementary variable areacella:
  /work/bd0854/DATA/ESMValTool2/CMIP6_DKRZ/CMIP/BCC/BCC-ESM1/1pctCO2/r1i1p1f1/fx/areacella/gn/v20190613/areacella_fx_BCC-ESM1_1pctCO2_r1i1p1f1_gn.nc
with files for supplementary variable sftlf:
  /work/bd0854/DATA/ESMValTool2/CMIP6_DKRZ/CMIP/BCC/BCC-ESM1/1pctCO2/r1i1p1f1/fx/sftlf/gn/v20190613/sftlf_fx_BCC-ESM1_1pctCO2_r1i1p1f1_gn.nc
```

Closes #2302

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [x] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#code-quality)
- [x] [🧪][2] and [🛠][1] [Documentation](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#documentation) is available
- [x] [🛠][1] [Unit tests](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#tests) have been added
- [x] [🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)
- [x] [🛠][1] Any changed [dependencies have been added or removed](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#dependencies) correctly
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

File: esmvalcore/_recipe/recipe.py [hunk 1]
```diff
@@ -250,25 +250,45 @@ def _add_to_download_list(dataset):
...
 
+def _get_files_str(dataset: Dataset) -> str:
+    """Get nice string representation of all files of a dataset."""
+    files_str = (
+        '\n'.join(
+            f'  {f}' if f.exists()  # type: ignore
+            else f'  {f} (will be downloaded)' for f in list(dataset.files)
+        )
+    )
+    return files_str
```
Comment (bouweandela) [inline esmvalcore/_recipe/recipe.py:289]:
```suggestion
    return '\n'.join(
        f'  {f}' if f.exists()  # type: ignore
        else f'  {f} (will be downloaded)' for f in dataset.files
    )
```
would be a bit more concise

File: esmvalcore/_recipe/recipe.py [hunk 2]
```diff
@@ -250,25 +250,45 @@ def _add_to_download_list(dataset):
...
     for dataset in datasets:
         _add_to_download_list(dataset)
         for supplementary_ds in dataset.supplementaries:
             _add_to_download_list(supplementary_ds)
 
-        files = list(dataset.files)
-        for supplementary_ds in dataset.supplementaries:
-            files.extend(supplementary_ds.files)
+
+def _log_all_files(datasets: Iterable[Dataset]) -> None:
```
Comment (bouweandela) [inline esmvalcore/_recipe/recipe.py:260]: Maybe rename to `_log_input_files`?

File: esmvalcore/_recipe/recipe.py [hunk 3]
```diff
@@ -250,25 +250,45 @@ def _add_to_download_list(dataset):
...
+    """Show list of files in log (including supplementaries)."""
+    for dataset in datasets:
+        # Only log supplementary variables if present
+        supplemantary_files_str = ""
+        if dataset.supplementaries:
+            for sup_ds in dataset.supplementaries:
+                supplemantary_files_str += (
```
Comment (bouweandela) [inline esmvalcore/_recipe/recipe.py:267]:
```suggestion
                supplementary_files_str += (
```

File: esmvalcore/_recipe/recipe.py (hunk 2 above)
Comment (schlunma) [reply esmvalcore/_recipe/recipe.py]: Done in https://github.com/ESMValGroup/ESMValCore/pull/2303/commits/69bfde561f106b905e1457196e822bc6b561a2a7

**Review by bouweandela — APPROVED** [review]

**Comment by schlunma** [comment]
Thanks for the review @bouweandela! @ESMValGroup/technical-lead-development-team does anyone have a minute for a final check and merge? Thanks!!

**Review by valeriupredoi — APPROVED** [review]
surprised Bouwe accepted printing stuff to screen :grin: Cheers, Manu :beer:

**Pull request merged**